docs/worker-versioning.md

```diff
@@ -0,0 +1,229 @@
+applied and the rest will be ignored.
+
+In the event that no assignment rule is applicable on a task (or the Task
+Queue is simply not versioned), the tasks will be dispatched to an
+unversioned Worker.
+
+### Redirect Rules
+Redirect rules should only be used when you want to move workflows and activities assigned to
+one Build ID (source) to another compatible Build ID (target). You are responsible to make sure
+the target Build ID of a redirect rule is able to process event histories made by the source
+Build ID by using [Patching](https://docs.temporal.io/workflows#patching) or other means.
+
+Most deployments are not expected to need these rules, however following
+ situations can greatly benefit from redirects:
+  - Need to move long-running Workflow Executions from an old Build ID to a
+    newer one.
+  - Need to hotfix some broken or stuck Workflow Executions.
+
+In steady state, redirect rules are beneficial when dealing with old
+Executions ran on now-decommissioned Build IDs:
+  - To redirecting the Workflow Queries to the current (compatible) Build ID.
+  - To be able to Reset an old Execution so it can run on the current
+    (compatible) Build ID.
+
+Redirect rules can be chained.
+
+## How to Use Worker Versioning
+To use Worker Versioning, you need to:
+1. Using Worker options to opt in Worker Versioning and pass Build ID. This is supported 
+in all SDKs. 
+
+Below is an example written in Go. It is assumed that the Build ID is provided via a env 
+variable.
+```go
+workerOptions := worker.Options{
+   BuildID: os.Getenv("BUILD_ID"),
+   UseBuildIDForVersioning: true,
+// ...
+}
+w := worker.New(c, "your_task_queue_name", workerOptions)
+```
+2. Deploy your worker with new Build ID while keeping the old Build ID still running.
+2. [Use CLI to update Worker Versioning Rules](#update-rules-using-cli) 
+to send new (and possibly existing, if you are deploying a compatible build) executions 
+to the new Build ID. Depending on your deployment strategy you may need to update the 
+rules multiple times.
+3. Use CLI to get reachability status of the old build ID. Once the Build ID is not 
+reachable, decommission the old Build ID.
+
+**Note for Temporal Server Operators:** Worker Versioning needs to be enabled from server
+before being able to run versioned workers or update the rules. To enable Worker Versioning
+set the following dynamic configs to `true`: `frontend.workerVersioningRuleAPIs`, 
+`frontend.workerVersioningWorkflowAPIs`. Both configs are per Task Queue.
+
+
+### Update Rules Using CLI
+You can use `temporal task-queue versioning` commands to update and read the Versioning 
+rules of a given Task Queue. Here are a few examples:
+
+**Add Assignment rule:** send 10% of new executions to Build ID "abc-123".
+```shell
+temporal task-queue versioning insert-assignment-rule --task-queue MY_TQ --build-id abc-123 --percentage 10
+```
+
+**Commit Build ID** to complete the rollout of "abc-123" and cleanup unnecessary rules.
```

Comment:
This is the first time you've mentioned the concept of "committing".  What does it do and when should I use it?
Is it safe to use this if I've got queries that might run on workflows assigned to older builds?